core-schema/src/main/resources/sql-schema/V8__Mod_clans.sql

```diff
@@ -86,3 +87,29 @@ CREATE FUNCTION clans_add_enemy
     INSERT INTO clans_clan_enemies (clan_id, enemy_id) VALUES (clan_identifier, enemy_identifier);
     RETURN 0;
   END;
+
+CREATE FUNCTION clans_create_new (name VARCHAR(32), leader_id INT) RETURNS TINYINT MODIFIES SQL DATA
+  COMMENT 'Returns 0 if method call successful. Returns 1 if name is not unique.
+  Returns 2 if user referenced by leader_id is already a clan member.'
+  BEGIN
+      -- Since they are all server-sided and therefore optimized by the engine AND there is no latency due to java-sql/io can i call
+    -- select queries as much as i fucking please? /s (but is this okay?)
+    IF EXISTS(SELECT 1 FROM clans_clan_info WHERE clan_name = name) THEN
+      -- i'm programming sql functions like i program java - i don't like try/catching when possible, so
+      -- why not just check explicitly for condition x/y - this one verifies explicitly uniqueness of clan name, so
+      -- i can return whatever "error code" i want.
```

Comment:
Right, you can return values as you wish.

I suggest you add an integration test to verify the java method which calls this function performs as expected. You can be surprised how easily the function implementation can go wrong and exhibit odd behavior. It's SQL, not Java.

Reply:
currently no methods use this sql function and i don't know why i haven't deleted it yet